gpflow/utilities/bijectors.py

```diff
@@ -5,17 +5,50 @@
 from .. import config
 from .utilities import to_default_float
 
+
 __all__ = ["positive", "triangular"]
 
 
-def positive(lower: Optional[float] = None):
-    lower_value = config.default_positive_minimum()
-    if lower_value is None:
-        return tfp.bijectors.Softplus()
+_POSITIVE_BIJECTOR_MAP = {
+    "exp": tfp.bijectors.Exp,
+    "softplus": tfp.bijectors.Softplus,
+}
+
+
+class Shift(tfp.bijectors.AffineScalar):
+    """Simple subclass so printed name is cleaner."""
+
+    def __init__(self, shift=None, validate_args=False, name='shift'):
+        super().__init__(shift=shift, validate_args=validate_args, name=name)
+
+
+def positive(lower: Optional[float] = None,
+             bijector: Optional[str] = None) -> tfp.bijectors.Bijector:
+    """
+    Returns a positive bijector (a reversible transformation from real to positive numbers).
+
+    :param lower: lower bound override (defaults to config.default_positive_minimum())
+    :param bijector: bijector method override (defaults to config.default_positive_bijector())
+    :returns: a bijector instance
+    """
+    bijector = _get_base_positive_bijector(bijector)
+    if lower is None:
+        lower = config.default_positive_minimum()
+    if lower is not None:
+        # Apply lower bound shift after applying base positive bijector
+        shift = Shift(shift=to_default_float(lower))
+        bijector = tfp.bijectors.Chain([bijector, shift])
```

Comment:
```suggestion
        bijector = tfp.bijectors.Chain([shift, bijector])
```

Shift should be first

Reply:
Actually I think bijector needs to be first - I believe that was the cause of the original bug in #1110. Here is a counterexample (in some kind of pseudo-code...):
```
unconstrained = -20.0
constrained = softplus(unconstrained + 1e-6)
constrained > 1e-6  # False!
```
whereas
```
unconstrained = -20.0
constrained = softplus(unconstrained) + 1e-6
constrained > 1e-6  # True
```

Reply:
Indeed, I checked documentation, https://www.tensorflow.org/probability/api_docs/python/tfp/bijectors/Chain. It is clearer what's happening. Good catch!

Reply:
It would be great if you could add this as a test!

Reply:
Test for what, that people read the documentation :) ?

Reply:
Soooo, turns out I was totally wrong. `Chain` applies the transformations in reverse order so the original implementation was correct - good job you suggested adding a test!